addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -200,19 +215,21 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 }
             }
         }
+
         if (!found && msg.getRequestHeader().getURI().getEscapedPath() != null) {
             // Handle jsessionid like:
             // http://tld.gtld/fred;jsessionid=1A530637289A03B07199A44E8D531427?foo=bar
             Matcher jsessMatcher = null;
+            String path = "";
             try {
-                jsessMatcher =
-                        Pattern.compile("jsessionid=[\\dA-Z]*", Pattern.CASE_INSENSITIVE)
-                                .matcher(msg.getRequestHeader().getURI().getPath());
+                // We lowercase the path because the regex has a specific session ID
+                // in it which can't be made case insensitive
+                path = msg.getRequestHeader().getURI().getPath().toLowerCase();
+                jsessMatcher = PATHSESSIONIDPATTERN.matcher(path);
             } catch (URIException e) {
+                e.printStackTrace();
```

Comment:
Don't print, if you really think this shouldn't be ignored log it instead (debug level).